### DISCUSSION TITLE
Multiple PRs for same dependency in the same file

### DISCUSSION THREAD

**Discussion by pabardz** [renovatebot/renovate#24599] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35

### If you're self-hosting Renovate, select which platform you are using.

Gitlab

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I am using the regex manager to update the same dependency used multiple time in the same file. Here the file read by RegexManager:

``` 
      - name: app-env1
        chartName: private-dependency
        chartVersion: 4.10.3

      - name: app-env2
        chartName: private-dependency
        chartVersion: 4.10.3

      - name: app-env3
        chartName: private-dependency
        chartVersion: 4.10.3
```

Here's the renovate configuration:

```
    additionalBranchPrefix: "{{packageFileDir}}-",
    regexManagers: [{
       fileMatch: ["^.*/argocd-config/values.yaml"],
       datasourceTemplate: "helm",
       matchStrings: [
           "chartName: (?<depName>.*?)\n(?:.*chartRepoURL: (?<registryUrl>.*?)\n)?.*chartVersion: (?<currentValue>.*)"
       ],
       registryUrlTemplate: "{{#if registryUrl}}{{registryUrl}}{{else}}https://xxxx{{/if}}"
    }],
```

Let's say the version 4.11.0 of the `private-dependency` is released, renovate will open 1 pull request with 3 changes.

My question is: Is it possible to force renovate to create 3 pull requests? One by change? 
I know it depends on the branch name, but in my case, the dependency is the same and it's in the same file. Is there a workaround for that? Can I add a custom variable linked to the name of my app somewhere to use that in the branch name?

So far the branch name is: `/renovate/live/argocd-config-private-dependency-4.x`

### What you tried so far.

I looked at the documentation and I think it's not possible to do that. 

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
They will go into different branches (PRs) if they have different `depName` values, because the branch uses `depName` by default. So you could perhaps use `app-env1/private-dependency` as `depName`, for example. You can use `depNameTemplate` to _compile_ the `depName` instead of extracting it directly in regex groups

**Reply by pabardz** [reply]
Oh thanks, I was not aware of the distinction between `depName` and `packageName`. 

I updated my configuration with:

```
       depNameTemplate: "{{#if depName}}{{packageName}}-{{depName}}{{else}}{{packageName}}{{/if}}",
       matchStrings: [
           "(?:.*platform: (?<depName>.*?)\n)?.*chartName: (?<packageName>.*?)\n(?:.*chartRepoURL: (?<registryUrl>.*?)\n)?.*chartVersion: (?<currentValue>.*)"
       ],
```

And I added a specific label in the dependencies file:

```
      - name: app-env1
        platform: env1
        chartName: private-dependency
        chartVersion: 4.10.3

      - name: app-env2
        platform: env2
        chartName: private-dependency
        chartVersion: 4.10.3
```

It works like a charm, thank you :)

**Reply by rarkins** [reply]
Thanks for sharing your solution!
